### PR TITLE
`Development`: Delete process polyfill and adjust paste-related e2e tests

### PR DESCRIPTION
**Pull request by pzdr7**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Changes affecting Programming Exercises
- [x] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).

### Motivation and Context

- In the Webpack 5 upgrade (#3470), support for the automatic polyfill of the process variable was [discontinued](https://webpack.js.org/migrate/5/#run-a-single-build-and-follow-advice) as it should be avoided in web environments. Nevertheless, we use the process dependency and manually polyfill this into the web environment. Note that we no longer use Webpack
- The Monaco editor uses `typeof process !== undefined` when checking if it is running in a native (rather than a web) environment. This broke pasting into the editor with CTRL+V on multiple browsers (Firefox was unaffected).
 
### Description

- Removes the `process` polyfill and its dependency. 
- Removes an import of 'process' from playwright
- Makes the paste mechanism used in Cypress more consistent (& platform-independent) with the playwright E2E tests.

### Steps for Testing

Prerequisites:
- Google Chrome or Microsoft Edge
- 1 Student
- 1 Programming exercise with the online code editor enabled

1. Log in to Artemis
2. Navigate to the programming exercise and open it in the online code editor
3. Open a file and ensure you can cut, copy, and paste using the keyboard shortcuts (usually Ctrl+X,Ctrl+C, and Ctrl+V).

#### Code Review
- [ ] Code Review 1
#### Manual Tests
- [ ] Test 1

## Summary by CodeRabbit

- **Refactor**
	- Improved the paste functionality in the Online Editor by changing the method used to insert text.
- **Chores**
	- Removed the `process` dependency across various configuration and script files, streamlining the project setup.

### CODE REVIEW

File: src/main/webapp/app/polyfills.ts [hunk 1]
```diff
@@ -1,8 +1,5 @@
 import 'zone.js';
 import '@angular/localize/init';
-import * as process from 'process';
 
 // Fix needed for SockJS, see https://github.com/sockjs/sockjs-client/issues/439
```
[the remark this answers is not in the document]
Comment (pzdr7) [reply src/main/webapp/app/polyfills.ts:4]: `process` is a node global and does not require an import

[the remark this answers is not in the document]
Comment (pzdr7) [reply src/main/webapp/app/polyfills.ts:4]: I have already reviewed the usages and found that they do not occur in the browser

**Review by b-fein — APPROVED** [review]
Code :+1:

**Review by krusche — APPROVED** [review]

**Review by coolchock — APPROVED** [review]
tested on Windows using Chrome and Edge and on Mac using Chrome and Safari. Works as expected 👍

**Pull request merged**